generators/client/templates/angular/src/main/webapp/app/shared/alert/alert-error.component.ts.ejs

```diff
@@ -106,21 +112,24 @@ export class <%=jhiPrefixCapitalized%>AlertErrorComponent implements OnDestroy {
         });
     }
 
-    setClasses(alert) {
-        return {
-            'jhi-toast': alert.toast,
-            [alert.position]: true
-        };
+    setClasses(alert: JhiAlert): { [key: string]: boolean } {
+        const classes = { 'jhi-toast': !!alert.toast };
+        if (alert.position) {
+            return { ...classes, [alert.position]: true };
+        }
+        return classes;
     }
-
-    ngOnDestroy() {
-        if (this.cleanHttpErrorListener !== undefined && this.cleanHttpErrorListener !== null) {
-            this.eventManager.destroy(this.cleanHttpErrorListener);
-            this.alerts = [];
```

Comment:
Why aren't we doing this anymore ?

Reply:
Assigning value to class member on destroy doesn't make sense. I believe that the idea was as I guessed in https://github.com/jhipster/generator-jhipster/pull/10901#issue-350746537 about `alert.component` `ngOnInit()`.

But `alert-error.component` actually is not tied to `JhiAlertService`, so this is really redundant, even `alertService.clear()` is not needed.

Reply:
Oh yes I didn't see the ngOnDestroy three lines up... Sorry 🤦‍♂